### DISCUSSION TITLE
:pipe does not work with nushell

### DISCUSSION THREAD

**Discussion by leetemil** [helix-editor/helix#8776] [Troubleshooting] [answered]
### Summary

Hi all, thanks for an amazing editor.
I think I've run into an issue: If Nushell/`nu` is used as shell, any `:pipe <cmd>` I have tried results in a `nu` error `Shell error: Error: nu::shell::file_not_found`. I assume that `nu` thinks `<cmd>` is a file and tries to open it (see logs below for full error).

If it's the case that Nushell is just not supported (yet), feel free to close this again.

### Reproduction Steps

 

I tried this:

1. Use Nushell as shell in whatever way that makes Helix use it for `:pipe`. I use it as default shell.
2. Open Helix and `:pipe` any command.

I expected this to happen:
The command would be executed by `nu`, like with `nu -c <cmd>`.

Instead, this happened:
File not found shell error.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
...
2023-11-09T22:07:14.971 helix_view::editor [ERROR] editor error: Shell error: Error: [31mnu::shell::file_not_found[0m

  [31m×[0m File not found
   ╭─[[36;1;4msource[0m:1:1]
 [2m1[0m │ nu
   · [35;1m▲[0m
   · [35;1m╰── [35;1mCould not access file 'sort': "No such file or directory (os error 2)"[0m[0m
   ╰────
...
```

</details>

### Platform

Linux

### Terminal Emulator

Alacritty

### Installation Method

pacman

### Helix Version

helix 23.10

**Comment by pascalkuthe** [comment, marked as answer]
It sounds like you just set shell to `nu` but it needs to be set to `nu -c`

**Reply by leetemil** [reply]
Thanks, that was the problem. With
```
[editor]
shell = ["nu", "-c"]
```
it works.

**Comment by chtenb** [comment]
shell = ["nu", "--stdin", "-c"]

That is what works for me
